keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/NetworkSamplesGenerator.java

```diff
@@ -25,9 +27,11 @@
     @Setter
     private int downSampleInterval = 1;
 
+    private ProgressBar progressBar;
 
-    public NetworkSamplesGenerator(SamplingAlgorithm algorithm) {
+    public NetworkSamplesGenerator(SamplingAlgorithm algorithm, ProgressBar progressBar) {
         this.algorithm = algorithm;
+        this.progressBar = progressBar;
     }
 
     public NetworkSamples generate(final int totalSampleCount) {
```

Comment:
Should we have a method on the progress bar to reset it and prepare it that's called at the start of generate() and stream()?  Theoretically there's nothing stopping someone calling generate multiple times on a single NetworkSampleGenerator (or even from calling generate and then stream() even...

Reply:
Good point. Since it actually creates a new thread, I'd prefer it to be considered dead when you call `finish()` and then create a new one when generate is called again. I've added a Supplier of a Progress bar to be passed here that way the generate method can create a new ProgressBar when it likes.